x-pack/plugins/observability/public/pages/slos/components/slo_list_search_filter_sort_bar.tsx

```diff
@@ -23,82 +23,93 @@ import React, { useEffect, useState } from 'react';
   {
-    label: i18n.translate('xpack.observability.slos.list.sortBy.budgetRemaining', {
-      defaultMessage: 'Budget remaining',
+    label: i18n.translate('xpack.observability.slos.list.sortBy.indicatorType', {
+      defaultMessage: 'Indicator type',
     }),
-    type: 'budgetRemaining',
+    type: 'indicator_type',
   },
 ];
 
-const STATUS_OPTIONS: SortItem[] = [
+const INDICATOR_TYPE_OPTIONS: Array<Item<FilterType>> = [
   {
-    label: i18n.translate('xpack.observability.slos.list.statusFilter.violated', {
-      defaultMessage: 'Violated',
+    label: i18n.translate('xpack.observability.slos.list.indicatorTypeFilter.apmLatency', {
+      defaultMessage: 'APM latency',
     }),
-    type: 'violated',
+    type: 'sli.apm.transaction_duration',
   },
   {
-    label: i18n.translate('xpack.observability.slos.list.statusFilter.healthy', {
-      defaultMessage: 'Healthy',
+    label: i18n.translate('xpack.observability.slos.list.indicatorTypeFilter.apmAvailability', {
+      defaultMessage: 'APM availability',
     }),
-    type: 'healthy',
+    type: 'sli.apm.transaction_error_rate',
+  },
+  {
+    label: i18n.translate('xpack.observability.slos.list.indicatorTypeFilter.customKql', {
+      defaultMessage: 'Custom KQL',
+    }),
+    type: 'sli.kql.custom',
   },
 ];
 
 export function SloListSearchFilterSortBar({
   loading,
   onChangeQuery,
   onChangeSort,
-  onChangeStatusFilter,
+  onChangeIndicatorTypeFilter,
 }: SloListSearchFilterSortBarProps) {
   const [isFilterPopoverOpen, setFilterPopoverOpen] = useState(false);
   const [isSortPopoverOpen, setSortPopoverOpen] = useState(false);
 
   const [sortOptions, setSortOptions] = useState(SORT_OPTIONS);
-  const [statusOptions, setStatusOptions] = useState(STATUS_OPTIONS);
+  const [indicatorTypeOptions, setIndicatorTypeOptions] = useState(INDICATOR_TYPE_OPTIONS);
 
   const selectedSort = sortOptions.find((option) => option.checked === 'on');
-  const selectedStatusFilters = statusOptions.filter((option) => option.checked === 'on');
+  const selectedIndicatorTypeFilter = indicatorTypeOptions.filter(
+    (option) => option.checked === 'on'
+  );
 
   const handleToggleFilterButton = () => setFilterPopoverOpen(!isFilterPopoverOpen);
   const handleToggleSortButton = () => setSortPopoverOpen(!isSortPopoverOpen);
 
-  const handleChangeSort = (newOptions: SortItem[]) => {
+  const handleChangeSort = (newOptions: Array<Item<SortType>>) => {
     setSortOptions(newOptions);
     setSortPopoverOpen(false);
   };
 
-  const handleChangeStatusOptions = (newOptions: SortItem[]) => {
-    setStatusOptions(newOptions);
-    setFilterPopoverOpen(false);
-
-    onChangeStatusFilter(newOptions.filter((option) => option.checked === 'on'));
+  const handleChangeIndicatorTypeOptions = (newOptions: Array<Item<FilterType>>) => {
+    setIndicatorTypeOptions(newOptions);
+    onChangeIndicatorTypeFilter(
+      newOptions.filter((option) => option.checked === 'on').map((option) => option.type)
+    );
```

Comment:
I'm not closing the popover anymore when selecting a filter, so you can select many and see the list refreshing